SFS_ir/interpolate_ir.m

```diff
@@ -1,33 +1,36 @@
-function [ir_new,x0_new] = interpolate_ir(ir,x0,xs,conf)
-%INTERPOLATE_IR interpolates three given IRs for the given angle
+function [ir_new,weights_new,x0_new] = interpolate_ir(ir,weights,x0,conf)
+%INTERPOLATE_IR interpolates the given impulse responses according to their weights
 %
-%   Usage: [ir_new,x0_new] = interpolate_ir(ir,x0,xs,conf)
+%   Usage: [ir_new,weights_new,x0_new] = interpolate_ir(ir,weights,x0,conf)
 %
 %   Input parameters:
-%       ir      - matrix containing impulse responses in the form [M C N], where
-%                     M ... Number of measurements (2<=M<=3)
-%                     C ... Number of channels
-%                     N ... Number of samples
-%       x0      - matrix containing positions of single impulse
-%                 responses [2 M] / (rad, rad)
-%       xs      - desired position after interpolation [2 1] / (rad, rad)
-%       conf    - configuration struct (see SFS_config)
+%       ir           - matrix containing impulse responses in the form [M C N], where
+%                          M ... Number of measurements
+%                          C ... Number of channels
+%                          N ... Number of samples
+%       weights      - M weights for impulse reponses
+%       x0           - M positions corresponding to given impulse responses
+%       conf         - configuration struct (see SFS_config)
 %
 %   Output parameters:
-%       ir_new  - impulse response for the given position [1 C N]
-%       x0_new  - position corresponding to the returned impulse response
+%       ir_new       - impulse response for the given position [1 C N]
+%       weights_new  - weights corresponding to impulse responses used for
+%                      interpolation
+%       x0_new       - position corresponding to impulse responses used for
+%                      interpolation
 %
-%   INTERPOLATE_IR(ir,x0,xs,conf) interpolates the two to three given impulse
-%   responses from ir with their corresponding angles x0 for the given angles
-%   xs and returns an interpolated impulse response.
-%	For the 1D case, the interpolation method differs depending on the setting
-%	of conf.ir.interpolationmethod:
+%   INTERPOLATE_IR(ir,x0,xs,conf) interpolates the given impulse responses by
+%   applying the given weights and returns the interpolated impulse response as
+%   well as its corresponding position. Only impulse responses with weights >= prec
+%   will be used.
```

Comment:
It is not obvious to the user what `prec` might be as it is not something that can be configured, but is hardcoded in the function itself. This is fine, but maybe you should replace `prec` here by its actual value `0.001`?